Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -180,22 +181,28 @@ function isNullOrEmpty(str) {
 }
 
 // Creates the string that comprises the build summary text, given the location of the /sonar build folder.
-function createSonarQubeBuildSummary(sqBuildFolder: string): string {
-    // Task report is not created for PR builds
+function createSonarQubeBuildSummary(sqBuildFolder:string):Q.Promise<string> {
+    // Task report is not created for PR builds - inform the user in the build summary
     if (isPrBuild()) {
+        tl.debug('Pull request build, will not attempt to fetch detailed SonarQube report');
         // Looks like: Detailed SonarQube reports are not available for pull request builds.
-        return tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild');
-    }
-
-    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
-    if (!taskReport) {
-        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
+        return Q.when(tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild'));
```

Comment:
Hmm, if Q.when behaves exactly like then(), won't it swallow errors? Wouldn't it be better to use https://github.com/kriskowal/q#using-qpromise instead?